server/service/osquery.go

```diff
@@ -589,6 +597,24 @@ func (svc *Service) GetDistributedQueries(ctx context.Context) (queries map[stri
 	return queries, discovery, accelerate, nil
 }
 
+// detectHostNotResponding returns whether the host hasn't been submitting results for sent queries.
+//
+// Notes:
+//   - We use `2 * interval`, because of the artificial jitter added to the intervals in Fleet.
+//   - Default values for:
+//     - host.DistributedInterval is usually 10s.
+//     - svc.config.Osquery.DetailUpdateInterval is usually 1h.
+func (svc *Service) detectHostNotResponding(host *fleet.Host) bool {
+	interval := svc.config.Osquery.DetailUpdateInterval
+	if time.Duration(host.DistributedInterval) > interval {
+		interval = time.Duration(host.DistributedInterval)
+	}
+	// The following means Fleet hasn't received the distributed/request from the host
+	// "in a while", thus we assume the host is having some issue in executing the
+	// queries or sending the results.
+	return svc.clock.Now().Sub(host.DetailUpdatedAt) > 2*interval
+}
```

Comment:
Now I wonder if we really need a new table and store such information if this is deduced from the values from `hosts` table (and config).

Maybe the result of `detectHostNotResponding` could be exposed APIs that return `Host`s?

(Sorry that I didn't see this sooner :/, please let me know if it makes sense, I may be missing something again :)

Reply:
@lucasmrod, that makes sense. I suppose it depends on how we expect to use the detections. In the places where we would care whether a host is non-responsive, would we already have the full record of the host and the config?  Are there reasons we might want to have a separate db path to look up if a host is not responding outside of the main `hosts` table?

Reply:
@lucasmrod  it seems like we should put the non-responsive hosts ticket on hold until we have a better idea of how the UI/fleetctl would utilize the information. At this point, I think all that would be left in this PR would be the `detectHostNotResponding` method, which wouldn't need to be called anywhere since we aren't persisting the information in the db and we don't have any specs for how to surface it via the UI/fleetctl. What do you think?